openvpn/middlewares/client/bytescount/selective_stats_handler.go

```diff
@@ -0,0 +1,22 @@
+package bytescount
+
+import (
+	"github.com/pkg/errors"
+	"time"
+)
+
+// NewIntervalStatsHandler creates and returns composite handler, which invokes internal handler at given interval
+func NewIntervalStatsHandler(handler SessionStatsHandler, clock func() time.Time, interval time.Duration) (SessionStatsHandler, error) {
+	if interval < 0 {
+		return nil, errors.New("Invalid 'interval' parameter")
+	}
+	var lastTime *time.Time = nil
```

Comment:
var lastTime = &time.Time{}. No need to check for nil, also it would be initialized to zero (unix style) and will trigger first call to handler. Maybe even pointer is not needed here.

Reply:
Added boolean var